web/src/components/Pretreatment.vue

```diff
@@ -1,160 +1,269 @@
-<script>
-import { SET_SELECTION } from '@/store/mutations.type';
-import { loadDataset } from '@/utils/mixins';
+<script lang="ts">
+import { defineComponent, inject, ref } from '@vue/composition-api';
+import VueRouter from 'vue-router';
+import { LOAD_DATASET } from '../store/actions.type';
+import { INTIALIZE_DATASET, SET_SELECTION } from '../store/mutations.type';
 import analyses from './vis/analyses';
+import store from '../store';
 
-export default {
-  mixins: [loadDataset],
+interface Dataset {
+  id: string;
+}
+
+interface Problem {
+  multi: boolean;
+  type: string;
+  context: string;
+  row_index?: string;
+  column_index?: string;
+}
+
+export default defineComponent({
   props: {
     problem: {
       type: String,
       default: null,
     },
     id: {
       type: String,
+      required: true,
       default: null,
     },
   },
-  data() {
-    return {
-      datasets: this.$store.state.datasets,
-      analyses,
-    };
-  },
-  methods: {
-    valid(dataset) {
-      return this.$store.getters.valid(dataset.id);
-    },
-    isMerged(dataset) {
-      return this.$store.getters.isMerged(dataset.id);
-    },
-    problemNav(problem, id) {
+  setup(props) {
+    // TODO Use injected router instead of importing to avoid circular import
+    const router = inject('router') as VueRouter;
```

Comment:
I recall that you asked recently how to get this inside a composition function: import or inject.  I don't remember what I said, but I went to look at the vue router next docs and inject will be preferred, but you'll be able to use builtin composition functions from the plugin like this:

https://next.router.vuejs.org/api/#composition-api

You may have already known that.  Anyway, this is fine with me.